lib/bap/bap_self.ml

```diff
@@ -60,4 +60,73 @@ module Create() = struct
   let warning f = message Warning ~section:name f
   let error f = message Error ~section:name f
 
+  module Config = struct
+    let plugin_name = name
+    include Bap_config
+
+    let (/) = Filename.concat
+    let confdir = confdir / plugin_name
+    let conf_filename = confdir / "config"
+
+    exception Improper_format of string
+
+    let get_env_options () =
```

Comment:
`Sys.getenv` does not get *all* the ones we want. It only returns value for a specific option.
In case we want to get the list of all config values passed to a plugin, we cannot just use `Sys.getenv`